src/sync/sync_session.hpp

```diff
@@ -149,7 +149,9 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     //
     // NOTE: This is intended for use only in very specific circumstances. Please check with the
     // object store team before using it.
+#if (REALM_VERSION_MAJOR >= 4)
```

Comment:
It should probably be guarded by the version of `realm-sync`, because the function is from there.

Reply:
Yes, a better check is `REALM_SYNC_VER_MAJOR > 1`.

Reply:
I will do it in a separate PR later.